### PR TITLE
allow use external typeface in native code

### PR DESCRIPTION
**Pull request by sospartan**
Expose method to implement changing font family cache.  Like @ide suggested in #4420 , this will helpful for using remote font file (use `Typeface#createFromFile` to load downloaded font file).   
iOS's CoreText  already allow this in native code.

### CODE REVIEW

**Comment by sospartan** [comment]
@mkonicek @satya164 any updates on this?

**Comment by satya164** [comment]
@sospartan Most of the team is on holiday. So it might take some time for this to be reviewed.

**Comment by sospartan** [comment]
@satya164 ok. Merry Christmas to you too.

**Comment by mkonicek** [comment]
cc @andreicoman11, @foghina

File: ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java [hunk 1]
```diff
@@ -72,6 +72,27 @@ public static ReactFontManager getInstance() {
...
+   * Add additional font family ,or replace the exist one in the font memory cache.
+   * @param style
+   * @see {@link Typeface#DEFAULT}
+   * @see {@link Typeface#BOLD}
+   * @see {@link Typeface#ITALIC}
+   * @see {@link Typeface#BOLD_ITALIC}
+   */
+  public void setTypeface(
+          String fontFamilyName,
+          int style,Typeface typeface){
```
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java]: typeface on new line

File: ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java [hunk 2]
```diff
@@ -72,6 +72,27 @@ public static ReactFontManager getInstance() {
...
+    if (typeface!=null) {
```
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java]: spaces around !=

File: ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java [hunk 3]
```diff
@@ -72,6 +72,27 @@ public static ReactFontManager getInstance() {
     return typeface;
   }
 
+  /**
+   * Add additional font family ,or replace the exist one in the font memory cache.
```
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java]: spacing around ,

**Comment by skevy** [comment]
Just wanted to ping on this @mkonicek -- we're using this internally @ Exponent and it's working fine. Would love for this to go through final review and merged.

**Comment by andreicoman11** [comment]
I'll take a look today.

**Comment by skevy** [comment]
Awesome @andreicoman11 gracias

**Comment by mkonicek** [comment]
Thanks for the ping @skevy.

**Pull request closed without merging**